src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java

```diff
@@ -65,6 +67,11 @@ public boolean executeOnce() {
             try (Socket ignored = new Socket(host, port)) {
                 return true;
             } catch (IOException e) {
+                ByteArrayOutputStream out = new ByteArrayOutputStream(); 
+                e.printStackTrace(new PrintStream(out));
+                String str = new String(out.toByteArray());
+                LOGGER.warn("This is a message from AppIDMan opening the socket didn't work because of: ");
+                LOGGER.warn(str);
```

Comment:
If you want to log an exception, pass it to the logger as an exception; that way the logger can decide whether or not to log anything at all and only bother calculating the stacktrace if the severity threshold is sufficient.

Reply:
Sorry i don't have any experience with the Logger in Java, so i don't really know how. 
Would this work ````LOGGER.warn(e)``` ?

Reply:
See https://github.com/jenkinsci/docker-plugin/blob/30a85b086ef6df57bd1bf776e88cbbf68eab7ab7/src/main/java/com/nirima/jenkins/plugins/docker/DockerJobProperty.java#L182 for an example.

Reply:
Thank you